src/main/scala/edu/berkeley/ce/rockslicing/Joint.scala

```diff
@@ -5,89 +5,121 @@ import breeze.linalg.{DenseMatrix, DenseVector}
+    * @param dip Dip direction of plane
+    * @return Tuple that contains x, y and z coordinates of point in global coordinates
+    */
+  private def localPointToGlobal(point: (Double, Double, Double), localOrigin: (Double, Double, Double),
+    normal: (Double, Double, Double), dip: Double): (Double, Double, Double) = {
+    val Nplane = if (normal._3 > -NumericUtils.EPSILON) {
+      // Ensures that normal will always point in -z global direction (ensures right-handed local coordinates)
+      -1.0 * DenseVector[Double](normal._1, normal._2, normal._3)
+    } else {
+      DenseVector[Double](normal._1, normal._2, normal._3)
+    }
+    val strike = (dip - math.Pi / 2) % (2 * math.Pi) // strike = dip - pi/2 (US convention)
+    val Nstrike = DenseVector[Double](math.cos(-strike), math.sin(-strike), 0.0)
+    val Ndip = linalg.cross(Nplane, Nstrike)
+
+    // Q defines the linear transformation to convert to global coordinates
+    val Q = DenseMatrix.zeros[Double](3,3)
+    Q(::, 0) := Nstrike
+    Q(::, 1) := Ndip
+    Q(::, 2) := Nplane
+
+    // Transform point from local to global coordinates
+    val transformedPoint = Q * DenseVector[Double](point._1, point._2, point._3)
+    (transformedPoint(0) + localOrigin._1, transformedPoint(1) + localOrigin._2, transformedPoint(2) + localOrigin._3)
+  }
+
+  /**
+    * Find a bounding sphere for a non-persistent joint. This function is not
+    * intended for use with persistent joints.
+    * @param normalVec The normal vector of the plane in which the joint lies
+    * @param distance The distance of the joint's plane from its local origin
+    * @param centerX The x coordinate of the joint's center
+    * @param centerY The y coordinate of the joint's center
+    * @param centerZ The z coordinate of the joint's center
+    * @param faces A sequence of faces specifying the joint's shape
+    * @param dip Dip direction of joint plane
+    * @return A pair where the first element is a triple giving the center of
+    *         the bounding sphere and the second element is the radius of the
+    *         bounding sphere.
+    */
   private def findBoundingSphere(normalVec: (Double,Double,Double), distance: Double, centerX: Double,
-      centerY: Double, centerZ: Double, faces: Seq[((Double,Double,Double),Double)]):
+      centerY: Double, centerZ: Double, faces: Seq[((Double,Double,Double),Double)], dip: Double):
       ((Double,Double,Double), Double) = {
     val basisVectors = Array(
-      Array[Double](1.0, 0.0, 0.0),
-      Array[Double](0.0, 1.0, 0.0),
-      Array[Double](0.0, 0.0, 1.0),
-      Array[Double](-1.0, 0.0, 0.0),
-      Array[Double](0.0, -1.0, 0.0),
-      Array[Double](0.0, 0.0, -1.0)
+      Array[Double](1.0, 0.0),
+      Array[Double](0.0, 1.0),
+      Array[Double](-1.0, 0.0),
+      Array[Double](0.0, -1.0)
     )
 
     val maxCoordinates = basisVectors.map { v =>
-      val linProg = new LinearProgram(3)
+      // val linProg = new LinearProgram(3)
```

Comment:
Can we remove this commented out line? You may also want to add a comment explaining why we only need two variables in the linear program now instead of three (since it's a 2D problem) for us math mortals. :)